ecl/eclagent/eclagent.cpp

```diff
@@ -2047,15 +2044,23 @@ void EclAgent::doProcess()
 
         if (getClusterType(clusterType)==ThorLCRCluster)
         {
-            if (w->getDebugValueBool("analyzeWhenComplete", agentTopology->getPropBool("@analyzeWhenComplete", defaultAnalyzeWhenComplete)))
+            if (getBoolWUOption(w, "analyzeWhenComplete", "analyzerOptions/@analyzeWhenComplete", defaultAnalyzeWhenComplete))
             {
-                switch (w->getState())
+                if (getBoolWUOption(w, "analyzeInEclAgent", "analyzerOptions/@analyzeInEclAgent", defaultAnalyzeInEclAgent))
```

Comment:
shouldn't these options be mutually exclusive? And if so, would be better if just 1 option, e.g. "analyzePerGraph"